File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
            
        for (int i = 0; i < n; i++) {
            int num = (arr1[i] | arr2[i]);
            String str = Integer.toBinaryString(num);
                        
            while (str.length() != n) {
                str = "0" + str;
            }
            
            answer[i] = str.replace('0', ' ').replace('1', '#');
        }
        return answer;
    }
}

File: 프로그래머스/lv1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    for i in range(n):
        num = bin(arr1[i] | arr2[i])[2:].zfill(n) # rjust(n, "0")도 된다. zfill()보다 더 범용적(?)이다.
        num = num.replace('0', ' ').replace('1', '#')
        answer.append(num)
    
    return answer